src/components/Homepage/LendByCategory/LoanCategoriesSection.vue

```diff
@@ -60,74 +68,92 @@ export default {
 			categoriesWithLoans: [],
 			activeCategory: null,
 			isLoggedIn: false,
-
-			// clientCategories: [],
-			// rowLazyLoadComplete: false,
-			// rightArrowPosition: undefined,
-			// leftArrowPosition: undefined,
+			scrollPos: 0,
 		};
 	},
-	apollo: {
-		query: lendByCategoryHomepageCategories,
-		preFetch(config, client) {
-			return client.query({
-				query: lendByCategoryHomepageCategories
-			}).then(({ data }) => {
-				// Get the array of channel objects from settings
-				const loanChannelsSetting = readJSONSetting(data, 'general.rows.value') || [];
-				// Get all channel ids for the row data
-				const channelIds = _map(loanChannelsSetting, 'id');
-				// Pre-fetch loan channel information, but no loans
-				return 	client.query({
-					query: loanChannelInfoQuery,
-					variables: {
-						ids: channelIds,
-					},
-					fetchPolicy: 'network-only',
-				});
-			});
-		},
-		result({ data }) {
-			this.isLoggedIn = _get(data, 'my.userAccount.id') !== undefined || false;
-
-			// Get the array of channel objects from settings,
-			// if successful set to categoryIds
-			const categorySettingsArray = readJSONSetting(data, 'general.rows.value');
-			if (categorySettingsArray) {
-				this.categoryIds = categorySettingsArray.map(setting => setting.id);
-			}
-
-			this.itemsInBasket = _map(_get(data, 'shop.basket.items.values'), 'id');
-
-			// Read the SSR ready loan channel info from the cache
-			try {
-				const categoryData = this.apollo.readQuery({
-					query: loanChannelInfoQuery,
-					variables: {
-						ids: this.categoryIds,
-					},
-				});
-				this.prefetchedCategoryInfo = _get(categoryData, 'data.lend.loanChannelsById') || [];
-			} catch (e) {
-				logReadQueryError(e, 'LoanCategoriesSection loanChannelInfoQuery');
-			}
-		}
-	},
 	created() {
-		// set initial active category
-		this.setActiveCategory(this.categoryIds[0]);
+		// TODO
+		// Get these queries in preFetch without causing an invariant error.
```

Comment:
I think this should either be guarded to only run in the browser, either with a window check or moved to mounted. I don't think there's anything in the created hook that requires this promise to complete on the server like there would be with a prefetch.

Maybe it's ok if the page only prefetches the featured loan section and this one loads up once in the browser. That would lighten the initial load.

Reply:
If we went this route of not prefetching the first set, I would consider implementing a more graceful loading state for area of the page. Something that could reveal and animate both the Names slider and the Loan rows as they come into view and transition between sets.

Requesting a larger amount of loans isn't out of the question... maybe only 12. Though having the dynamic need based loader would be better at some point.

Reply:
I moved the initialization code to mounted and added a loader. I think this is good enough to merge and continue progress. I am aware that in the case where we fetch 8 loans for a category, and some of those loans are filtered out the category row slider will have an insufficient amount of loans